test/browser/getOwnerDocument.test.js

```diff
@@ -0,0 +1,25 @@
+import { createElement, render } from 'preact';
+import { setupScratch, teardown } from '../_util/helpers';
+
+/** @jsx createElement */
+
+describe('parentDom.ownerDocument', () => {
+	/** @type {HTMLDivElement} */
+	let scratch;
+
+	before(() => {
+		scratch = setupScratch();
+	});
+
+	after(() => {
+		teardown(scratch);
+	});
+
+	it('should reference the correct document from the parent node', () => {
```

Comment:
Wouldn't this test pass without your change? I think this effectively asserts that `document.createElement('div').ownerDocument == document`. Can we write a test that mimics the original issue this PR fixes? I think it involves iframes or something.

Reply:
Yeah, a test for this could spy on `iframeDoc.createElement()` to assert that it is called instead of `document.createElement()`.

Reply:
Looks like the last changes to use a spy on `createTextNode` should resolve this review comment.

Hoping to see this PR merge into an official release soon!